samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -298,22 +298,25 @@ private JobCoordinator createJobCoordinator() {
   private boolean stopSamzaContainer() {
     boolean hasContainerShutdown = true;
     if (container != null) {
-      if (!container.hasStopped()) {
-        try {
-          container.shutdown();
-          LOGGER.info("Waiting {} ms for the container: {} to shutdown.", taskShutdownMs, container);
-          hasContainerShutdown = containerShutdownLatch.await(taskShutdownMs, TimeUnit.MILLISECONDS);
-        } catch (IllegalContainerStateException icse) {
```

Comment:
ICSE is not a timeout. Maybe set containerException to this value, or log this and then set hasContainerShutdown = false?

Reply:
I guess the idea is to make container shutdown idempotent? If so, removing ICSE is fine.